build/Dockerfile.dev

```diff
@@ -0,0 +1,17 @@
+FROM registry.access.redhat.com/ubi9/go-toolset:1.19 as build
```

Comment:
I created a dev dockerfile for improving the go modules caching and for future improvments such as hot reloading support.